### PR TITLE
Conduit: auto configuration generation with init command

### PR DESCRIPTION
**Pull request by mciccarello**
## Summary

Conduit configuration file can now be generated based on parameters passed in on the command line for importer, exporter and processors

## Test Plan

Unit test covered. Tested manually in various scenarios.

### CODE REVIEW

File: cmd/conduit/main.go [hunk 1]
```diff
@@ -144,15 +162,68 @@ func runConduitInit(path string) error {
...
-	_, _ = f.WriteString(sampleConfig)
+	var processorList []string
+
+	if processorsFlag != "" {
+		processorList = strings.Split(processorsFlag, ",")
+	}
+
+	var importer string
+	if importerFlag == "" {
+		importerFlag = "algod"
```
Comment (AlgoStephenAkiki) [inline cmd/conduit/main.go:173]: Reminder to change this to sync_algod when it gets merged.

**Review by AlgoStephenAkiki — APPROVED** [review]

**Review by winder — CHANGES_REQUESTED** [review]
Some minor suggestions.

File: cmd/conduit/main.go (hunk 1 above)
Comment (mciccarello) [reply cmd/conduit/main.go:173]: Stephen, can you explain?

File: cmd/conduit/conduit.test.init.default.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Generated conduit configuration file.
```
[the remark this answers is not in the document]
Comment (mciccarello) [reply cmd/conduit/conduit.test.init.default.yml:1]: It's used in a test case.

File: cmd/conduit/main_test.go [hunk 3]
```diff
@@ -16,23 +17,28 @@ import (
 	"github.com/algorand/indexer/loggers"
 )
 
+// TestInitDataDirectory tests the initialization of the data directory//go:embed conduit.test.init.default.yml
+
```
[the remark this answers is not in the document]
Comment (winder) [reply cmd/conduit/main_test.go:21]: Could you move this comment to the right spot and remove the trailing go:embed?

File: cmd/conduit/conduit.test.init.default.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (winder) [reply cmd/conduit/conduit.test.init.default.yml:1]: I see, I was confused by the duplicate comment and trailing go:embed mentioned in another comment I left

File: cmd/conduit/main.go [hunk 4]
```diff
@@ -161,19 +232,25 @@ func runConduitInit(path string) error {
...
-			return runConduitInit(data)
+			return runConduitInit(data, importer, processors, exporter)
 		},
 		SilenceUsage: true,
 	}
 
 	cmd.Flags().StringVarP(&data, "data", "d", "", "Full path to new data directory. If not set, a directory named 'data' will be created in the current directory.")
-
+	cmd.Flags().StringVarP(&importer, "importer", "i", "", "data importer name.")
+	cmd.Flags().StringVarP(&processors, "processors", "p", "", "comma-separated list of processors.")
```
[the remark this answers is not in the document]
Comment (mciccarello) [reply cmd/conduit/main.go]: Done

File: cmd/conduit/main_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (mciccarello) [reply cmd/conduit/main_test.go]: Removed

File: conduit/plugins/importers/filereader/sample.yaml [hunk 5]
```diff
@@ -1,14 +1,14 @@
...
+    block-dir: "/path/to/directory"
+    # RetryDuration controls the delay between checks when the importer has
+    # caught up and is waiting for new blocks to appear.
+    retry-duration: "5s"
+    # RetryCount controls the number of times to check for a missing block
+    # before generating an error. The retry count and retry duration should
+    # be configured according the expected round time.
+    retry-count: 5
+    # FilenamePattern is the format used to find block files. It uses go string formatting and should accept one number for the round.
+    filename-pattern: "%[1]d_block.json"
```
Comment (shiqizng) [inline conduit/plugins/importers/filereader/sample.yaml:14]: nit: do we need space before these lines?

Comment (mciccarello) [reply conduit/plugins/importers/filereader/sample.yaml:14]: For it to be formatted correctly we do.

**Review by shiqizng — APPROVED** [review]

**Comment by winder** [comment]
Moved to https://github.com/algorand/indexer/pull/1474

**Pull request closed without merging**